publications/tasks.py

```
@@ -7,228 +7,239 @@
-                    if isinstance(start_time, list):
-                        period_start = [d for d in start_time if d]
-                    if isinstance(end_time, list):
-                        period_end = [d for d in end_time if d]
-                except Exception as time_err:
-                    logger.error("Time period extraction failed for URL %s: %s", identifier_value, time_err)
-
-            except Exception as fetch_err:
-                logger.error("Error fetching HTML for %s: %s", identifier_value, fetch_err)
-                geom_object = GeometryCollection()
-                period_start, period_end = [], []
-
-            publication = Publication(
-                title=title_value,
-                abstract=abstract_text,
-                publicationDate=date_value,
-                url=identifier_value,
-                doi=doi_text,
-                source=journal_value,
-                geometry=geom_object,
-                timeperiod_startdate=period_start,
-                timeperiod_enddate=period_end,
-                job=event
+                g = extract_geometry_from_html(soup)
+                if g:
+                    geom = g
+            
+                if src and getattr(src, "is_preprint", False) and geom.empty:
+                    try:
+                        loc = Nominatim(user_agent="optimap-tasks").geocode(
+                        src.homepage_url or src.url
+                    )
+                        if loc:
+                            geom = Point(loc.longitude, loc.latitude)
+                    except Exception as e:
+                        logger.debug(
+                            "Preprint geocode failed for %s: %s",
+                            src.name if src else identifier,
+                            e
+                        )
+        
+                ps_list, pe_list = extract_timeperiod_from_html(soup)
+            except Exception:
```

Never silently catch exceptions.